persistence/test/setup_test.go

```diff
@@ -80,17 +83,19 @@ func NewTestPostgresContext(t testing.TB, height int64) *persistence.PostgresCon
 
 // TODO(olshansky): Take in `t testing.T` as a parameter and error if there's an issue
 func newTestPersistenceModule(databaseUrl string) modules.PersistenceModule {
-	// HACK: See `runtime/test_artifacts/generator.go` for why we're doing this to get deterministic key generation.
-	os.Setenv(test_artifacts.PrivateKeySeedEnv, "42")
-	defer os.Unsetenv(test_artifacts.PrivateKeySeedEnv)
-
-	cfg := runtime.NewConfig(&runtime.BaseConfig{}, runtime.WithPersistenceConfig(&types.PersistenceConfig{
-		PostgresUrl:    databaseUrl,
-		NodeSchema:     testSchema,
-		BlockStorePath: "",
-		TxIndexerPath:  "",
-		TreesStoreDir:  "",
-	}))
+	teardownDeterministicKeygen := keygenerator.GetInstance().SetSeed(42)
```

Comment:
This interface looks very clean IMO, so do we even need #414? Did you have something else in mind?

Reply:
I have created #414 only to track the individual todo-item in #361 (I have used the feature "convert to issue") so that when this PR is merged, the todo item is automatically marked as completed and there are also breadcrumbs to here basically.